File: common/src/main/java/com/jon/common/utils/DataFormat.java
package com.jon.common.utils;

import android.content.SharedPreferences;

public enum DataFormat {
    XML("XML"),
    PROTOBUF("Protobuf");

    private final String format;

    DataFormat(String format) {
        this.format = format;
    }

    public String get() {
        return format;
    }

    public static DataFormat fromPrefs(SharedPreferences prefs) {
        final String formatString = PrefUtils.getString(prefs, Key.DATA_FORMAT);
        return fromString(formatString);
    }

    public static DataFormat fromString(String formatString) {
        for (DataFormat format : values()) {
            if (format.get().equals(formatString)) {
                return format;
            }
        }
        throw new IllegalArgumentException("Unknown data format: " + formatString);
    }
}

File: common/src/main/java/com/jon/common/utils/DataFormat.kt
package com.jon.common.utils

import android.content.SharedPreferences
import com.jon.common.prefs.CommonPrefs
import com.jon.common.prefs.getStringFromPair

enum class DataFormat(private val format: String) {
    XML("XML"),
    PROTOBUF("Protobuf");

    override fun toString(): String {
        return format
    }

    companion object {
        fun fromPrefs(prefs: SharedPreferences): DataFormat {
            return fromString(
                    prefs.getStringFromPair(CommonPrefs.DATA_FORMAT)
            )
        }

        fun fromString(formatString: String): DataFormat {
            return values().first { it.toString() == formatString }
        }
    }
}
